api/secret.go

```diff
@@ -316,5 +324,14 @@ func ParseSecret(r io.Reader) (*Secret, error) {
 		return nil, err
 	}
 
+	// If the data is null, add raw data to secret data if present
+	if secret.Data == nil {
+		data := make(map[string]interface{})
+		if err := jsonutil.DecodeJSONFromReader(&teebuf, &data); err != nil {
+			return nil, err
+		}
+		secret.Data = data
+	}
```

Comment:
I think you can do something like this:

```suggestion
	// If the data is null, add raw data to secret data if present
	if secret.Data == nil {
		data := make(map[string]interface{})
		if err := mapstructure.Decode(secret, &data); err != nil {
			return nil, err
		}
		secret.Data = data
```

and then we don't need a second copy of the stream above. I think this might be a good optimization because it prevents duplicating the stream every time we read, regardless of whether we have a `Data` field.  In this case, we'll only need to make a copy of the data on-demand.

How do you feel about this?